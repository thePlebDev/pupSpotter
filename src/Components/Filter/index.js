import React,{useState,useRef,useEffect} from 'react';
import styled from 'styled-components'
import SearchIcon from '@material-ui/icons/Search';
import useOutsideClick from '../../Hooks/UseOutsideClick';
import axios from 'axios';
// import axiosGet from '../../Utils/axiosGet.js'
import Button from '../Button'

import {backendUrl} from '../../utils/Constants';
import {SideBar,Tab} from '../../Assets/SideBarStyles';
import filterFuncs from '../../utils/FilterFuncs';

const RightSideBar =styled(SideBar)`
right:0;
transform:${props => props.show? 'translateX(0px)' : 'translateX(240px)'};
`

const Text = styled.h3`
  margin-left:43%;
  color:white;
`

const RightTab = styled(Tab)`
right:200px;
border-radius:10%;
&:hover{
  right:205px;
}
`


const Filter = (props)=>{
  const node = useRef(null);
  const {show,setShow} = useOutsideClick(node)

  return(

    <RightSideBar show={show} data-testid="filter">
      <RightTab onClick={()=>setShow(!show)} data-testid="onClick">
        <SearchIcon ref={node} style={{fontSize:'60',color:'white',cursor:'pointer'}}/>
      </RightTab>
      <Text>Filters:</Text>
      <Button data-testid="button" handleClick={filterFuncs.filter} url={'spot/all'} setData={props.setMapData} words='Reset Filters'/>
      <Button data-testid="button" handleClick={filterFuncs.filter} url={'spot/highest'} sSToken={'highest'} setData={props.setMapData} words='Best Boy'/>
      <Button data-testid="button" handleClick={filterFuncs.filter} url={'spot/lowest'} sSToken={'lowest'} setData={props.setMapData} words='Worst Boy'/>
    </RightSideBar>
  )
}


export default Filter
