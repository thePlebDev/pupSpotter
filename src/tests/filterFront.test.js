import React from 'react';
import Filter from '../Components/Filter';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';


Enzyme.configure({
  adapter: new EnzymeAdapter()
})

describe('testing the filter component',()=>{
  it('should test the first snapshot',()=>{
    const tree = renderer.create(<Filter/>).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('should render properly',()=>{
    const wrapper = shallow(<Filter/>)
    expect(wrapper.find('[data-testid="filter"]').length).toBe(1)
    expect(wrapper.find('[data-testid="button"]').length).toBe(3)
  })

})
