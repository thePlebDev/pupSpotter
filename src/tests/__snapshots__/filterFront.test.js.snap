// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the filter component should test the first snapshot 1`] = `
<div
  className="sc-AxirZ sc-AxhCb cCkLUO"
  data-testid="filter"
>
  <div
    className="sc-AxiKw sc-AxgMl gluXKe"
    data-testid="onClick"
    onClick={[Function]}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      style={
        Object {
          "color": "white",
          "cursor": "pointer",
          "fontSize": "60",
        }
      }
      viewBox="0 0 24 24"
    >
      <path
        d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
      />
    </svg>
  </div>
  <h3
    className="sc-AxhUy kFIZtd"
  >
    Filters:
  </h3>
  <button
    className="sc-AxjAm jvYtuW"
    data-testid="button"
    id="button"
    onClick={[Function]}
  >
    Reset Filters
  </button>
  <button
    className="sc-AxjAm jvYtuW"
    data-testid="button"
    id="button"
    onClick={[Function]}
  >
    Best Boy
  </button>
  <button
    className="sc-AxjAm jvYtuW"
    data-testid="button"
    id="button"
    onClick={[Function]}
  >
    Worst Boy
  </button>
</div>
`;
